onnxmltools/convert/sklearn/operator_converters/KNN.py

```diff
@@ -133,12 +152,101 @@ def convert_sklearn_knn(scope, operator, container):
+        for i in range(len(classes)):
+            container.add_node('Equal', [labels_name[i], topk_labels_name],
+                                output_label_name[i])
+            # Casting to Int32 instead of Int64 as ReduceSum doesn't seem to support Int64 
+            apply_cast(scope, output_label_name[i], output_cast_label_name[i], container,
+                       to=onnx_proto.TensorProto.INT32)
+            container.add_node('ReduceSum', output_cast_label_name[i],
+                                output_label_reduced_name[i], axes=[1])
+
+        container.add_node('Concat', [s for s in output_label_reduced_name],
+                           concat_labels_name, name=scope.get_unique_operator_name('Concat'), axis=0)
+        container.add_node('ArgMax', concat_labels_name, 
+                           predicted_label_name, name=scope.get_unique_operator_name('ArgMax'))
+        if class_type == onnx_proto.TensorProto.INT32:
+            container.add_node('ArrayFeatureExtractor', [classes_name, predicted_label_name], final_label_name,
+                               name=scope.get_unique_operator_name('ArrayFeatureExtractor'), op_domain='ai.onnx.ml')
+            apply_reshape(scope, final_label_name, reshaped_final_label_name, container, desired_shape=[-1,])
+            apply_cast(scope, reshaped_final_label_name, operator.outputs[0].full_name, container,
+                       to=onnx_proto.TensorProto.INT64)
+        else:
+            container.add_node('ArrayFeatureExtractor', [classes_name, predicted_label_name],
+                               operator.outputs[0].full_name, 
+                               name=scope.get_unique_operator_name('ArrayFeatureExtractor2'), op_domain='ai.onnx.ml')
+
+        # Calculation of class probability
+        pred_label_shape = [-1, 1]
+
+        pred_label_shape_name = scope.get_unique_variable_name('pred_label_shape')
+        cast_pred_label_name = scope.get_unique_variable_name('cast_pred_label')
+        reshaped_pred_label_name = scope.get_unique_variable_name('reshaped_pred_label')
+        reduced_prob_name = scope.get_unique_variable_name('reduced_prob')
+
+        container.add_initializer(pred_label_shape_name, onnx_proto.TensorProto.INT64,
+                                  [len(pred_label_shape)], pred_label_shape)
+        ohe_model = OneHotEncoder(categorical_features='all').fit(training_labels.reshape((-1, 1)))
+
+        container.add_node('Reshape', [topk_labels_name, pred_label_shape_name], 
+                           reshaped_pred_label_name, name=scope.get_unique_operator_name('Reshape'))
+        apply_cast(scope, reshaped_pred_label_name, cast_pred_label_name, container, to=onnx_proto.TensorProto.INT64)
+
+        op_type = sklearn_operator_name_map[type(ohe_model)]
+        ohe_operator = Oper(ohe_model, cast_pred_label_name, op_type)
+        ohe_result_name = convert_sklearn_one_hot_encoder(scope, ohe_operator, container, flag=True)
```

Comment:
Humm, it's a very cool idea of code re-using but I would suggest not to do this because the one-hot-encoder here is trivial while the conversion code of scikit-learn's one-hot encoder is super difficult. For example, I believe you can just do
```
add_node('OneHotEncoder', cast_pred_label_name, ohe_result_name, name=scope.get_unique_operator_name('OneHotEncoder'), cats_int64s=list(range(len(classes)))
```